### PR TITLE
Support non blocking send on client

### PR DESCRIPTION
**Pull request by davidfowl**
https://github.com/aspnet/SignalR/issues/589

### CODE REVIEW

**Comment by moozzyk** [comment]
You changed only client. Shouldn't the server be changed as well to not send a response if this is a nonBlocking request? No one is waiting for this response and you will likely receive a [warning](https://github.com/aspnet/SignalR/blob/f8d91b54d4b677b619d716f5003f2ab98b614829/src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs#L178) for each such response.

**Comment by davidfowl** [comment]
Why would that warning ever happen? The sever never sends a message back for non blocking invocations and the client never sends a response back to the sever for a non blocking invocation.

**Comment by anurse** [comment]
I don't think the server honors `nonBlocking` right now. It'll send a completion anyway, which is a protocol error (warning, in this case)

**Comment by moozzyk** [comment]
What @anurse said.

**Comment by davidfowl** [comment]
I don't understand.

**Comment by davidfowl** [comment]
I do now 😄

File: src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs [hunk 1]
```diff
@@ -101,29 +101,46 @@ public void On(string methodName, Type[] parameterTypes, Func<object[], Task> ha
...
+            return InvokeCore(methodName, irq, args, nonBlocking: true);
+        }
+
+        private Task InvokeCore(string methodName, InvocationRequest irq, object[] args, bool nonBlocking)
         {
             ThrowIfConnectionTerminated();
-            _logger.LogTrace("Preparing invocation of '{target}', with return type '{returnType}' and {argumentCount} args", methodName, irq.ResultType.AssemblyQualifiedName, args.Length);
+            if (nonBlocking)
+            {
+                _logger.LogTrace("Preparing invocation of '{target}' and {argumentCount} args", methodName, irq.ResultType.AssemblyQualifiedName, args.Length);
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs]: super nit: Preparing non-blocking...

**Pull request merged**